plugins/device/rest.go

```diff
@@ -121,6 +123,58 @@ func deviceListHandler(w http.ResponseWriter, r *http.Request) {
 
 }
 
+func deviceEditHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	peerID := mux.Vars(r)["peerid"]
+	if uuid.Parse(peerID) == nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "invalid peer-id passed in url")
+		return
+	}
+
+	req := new(deviceapi.EditDeviceReq)
+	if err := restutils.UnmarshalRequest(r, req); err != nil {
+		logger.WithError(err).Error("Failed to Unmarshal request")
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrJSONParsingFailed)
+		return
+	}
+
+	deviceState, exists := deviceapi.DeviceStates[strings.ToLower(req.State)]
```

Comment:
You must disallow arbitrary mix of small and big caps - `eNaBlEd` or `enabLED` etc

Check if the user input is `enabled` or `disabled` and use that for validity. Keep it simple.